public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -195,6 +195,11 @@ export function createDashboardSceneFromDashboardModel(oldModel: DashboardModel)
     variables = new SceneVariableSet({
       variables: variableObjects,
     });
+  } else {
```

Comment:
If it is mandatory to get always a valid value in `variables`, probably it is a good idea to add a test for it. WDYT?

Reply:
Good idea, I have added the test :)

Reply:
yeah, +1 to ivan's. Overall what we see is it's better to attach even empty objects rather than conditonally adding theme. It simplifies downstream code a lot.